governance/reviews/cloudevents-08-29-2023.md

```diff
@@ -0,0 +1,210 @@
+| Contributor Ladder | Partial | [GOVERNANCE](https://github.com/cloudevents/spec/blob/main/docs/GOVERNANCE.md) | Several roles exist, but qualifications and advancement are not explained |
+| Maintainer Lifecycle | Missing | | How to become, or retire, a Voting Member is not explained anywhere |
+| Decision-making | Complete | [GOVERNANCE](https://github.com/cloudevents/spec/blob/main/docs/GOVERNANCE.md), [SDK Governance](https://github.com/cloudevents/spec/blob/main/docs/SDK-GOVERNANCE.md) | Nonlinear, but explained in exacting detail |
+| Code and Docs Ownership | Complete | [OWNERS](https://github.com/cloudevents/spec/blob/main/OWNERS), plus each repo | |
+| Security Reporting and response | Missing | | A security team exists but is not documented |
+| Communication and Meetings | Partial | [GOVERNANCE](https://github.com/cloudevents/spec/blob/main/docs/GOVERNANCE.md) | meeting process, no calendar.  should link to SDK comms info |
+
+#### Sub-projects, plugins, and related
+
+<!--- If the project has subprojects, plugins, or other divisions define them here. For each, is ownership and operation of clearly described? Are any standing committees/teams fully described, including listing their members? Does it conform to, align, and is it within scope of the governance expectations of the project? --->
+
+The project includes the following sub-projects, plugins, and other notable divisions:
+
+| Area | Ownership and Operation | Standing Bodies | Project Alignment | Notes |
+|:-----|:-----------------------:|:---------------:|:------------------|:---|
+| SDK-Java | Partial | Complete | Partial | Currently a WIP project |
+| SDK-Csharp | Partial | Complete | Partial | |
+| SDK-Javascript | Partial | Complete | Partial | |
+| SDK-Rust | Missing | Complete | Partial | No maintainers listed |
+| SDK-Powershell | Missing | Missing | Partial | No maintainers, no meetings or channel |
+| SDK-PHP | Partial | Complete | Partial | No maintainers listed |
+| SDK-Ruby | Partial | Complete | Partial | No maintainers listed |
+
+Language-specific Software Development Kits (SDKs) exist as subprojects of CloudEvents.  These are largely autonomous of the main spec project in a governance sense.  There is a [template for SDK governance](https://github.com/cloudevents/spec/blob/main/docs/SDK-GOVERNANCE.md) and other guidance docs, but SDK projects are largely left on their own after creation.  This leads to some problems:
+
+* Some SDKs have gone completely unmaintained
+* There are no processes for SDK archiving or SDK Maintainer offboarding
+* No SDK roles are defined other than "maintainer"
+* Certification of SDKs is not up to date
+* SDK repos do not appear to have OWNERS files or other administrative automation
+* It's impossible to tell if SDKs are well governed since they have no minutes or video of their monthly meetings
+* There's no defined governance relationship between other bodies and the SDKs or between maintainers of different SDKs
```

Comment:
Which "other bodies" are you thinking of?
Do we have to have a relationship across SDKs?

Reply:
Admins and voting members.

Reply:
Can you elaborate on what you're looking for beyond what in this section: https://github.com/cloudevents/spec/blob/main/docs/GOVERNANCE.md#membership

Reply:
You don't have to have a relationship, but if you don't you want to spell it out:  "SDKs are completely autonomous from other groups."  There's some practical considerations to declaring that though, and it might not be acceptable to the TOC.  For example:

- how do SDKs get removed/archived because of abandonment?
- who verifies that SDKs comply with the spec?
- who handles security reports against SDKs?

Reply:
I've opened https://github.com/cloudevents/spec/pull/1229 to address this.